avocado/core/multiplexer.py

```diff
@@ -408,8 +409,10 @@ def get_number_of_tests(self, test_suite):
         """
         # Currently number of tests is symetrical
         if self.variants:
-            return (len(test_suite) *
-                    sum(1 for _ in self.variants))
+            no_variants = sum(1 for _ in self.variants)
```

Comment:
This is one of the very few bad things about iterators! But the `sum` comprehension looks fine.